web/ui/mantine-ui/src/pages/query/HistogramChart.tsx

```diff
@@ -0,0 +1,309 @@
+import React, { FC } from "react";
+import { Histogram } from "../../types/types";
+import {
+  calculateDefaultExpBucketWidth,
+  findMinPositive,
+  findMaxNegative,
+  findZeroAxisLeft,
+  showZeroAxis,
+  findZeroBucket,
+  bucketRangeString,
+} from "./HistogramHelpers";
+import classes from "./HistogramChart.module.css";
+import { Tooltip } from "@mantine/core";
+
+interface HistogramChartProps {
+  histogram: Histogram;
+  index: number;
+  scale: string;
+}
+
+const HistogramChart: FC<HistogramChartProps> = ({
```

Comment:
I still noticed one display glitch when switching to linear mode, with a vertical grid line appearing left of my Y axis:

![image](https://github.com/prometheus/prometheus/assets/538008/c4a01fe0-d2c7-4995-98c0-d494f1e45493)

Reply:
Ah I might know whats causing that

I'll try to fix

Reply:
Hmm I actually can't seem to recreate this but my guess is that its the the y-axis marker for 0

This might be a stretch: but if you use the server from `main` with latest changes and run the new ui detached, does that help remove it?

Reply:
Huh strange, at the moment I can't reproduce it either anymore, no matter if I use the server from `main` or your PR branch. Maybe it was super data-dependent, but since it happened for all the sub-histograms, that would be weird too.

I guess I'll just keep an eye on it in the future in case it comes back.